src/url-utils.ts

```diff
@@ -30,6 +30,19 @@ const sharedRelativeUrlProperties =
 export type UrlString = string;
 
 /**
+ * Returns a URL with the basename removed from the pathname.  Strips the
+ * search off of the URL as well, since it will not apply.
+ */
+export function debaseUrl(href: UrlString): UrlString {
+  const u = url.parse(href);
+  if (typeof u.pathname === 'string') {
```

Comment:
What other types might pathname be? If it's just null/undefined, recommend instead 

```typescript
if (u !== null) {
  ...
}
```

I'd prefer that style because it documents for the reader that u.pathname is either a string or null/undefined (otherwise typescript would warn).